src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,163 @@
+package duke;
+
+import duke.commands.Command;
+import duke.commands.DoneCommand;
+import duke.commands.DeadlineCommand;
+import duke.commands.ByeCommand;
+import duke.commands.TodoCommand;
+import duke.commands.ListAllCommand;
+import duke.commands.DeleteCommand;
+import duke.commands.EventCommand;
+import duke.commands.FindCommand;
+
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+
+/**
+ * Class that represents a Parser object for parsing Strings to Commands
+ */
+public class Parser {
+
+    /**
+     * Returns a Command by parsing a user provided String input
+     * that can be executed.
+     *
+     * @param fullCommand A String input direct from the user.
+     * @return A subtype of Command
+     * @throws DukeException In the event of parse or invalid argument errors
+     */
+    public static Command parse(String fullCommand) throws DukeException {
+        String firstToken = fullCommand.split(" ")[0];
+
+        // Check if fullCommand is "bye"
+        if (fullCommand.equals("bye")) {
+            return new ByeCommand();
+        } 
+
+        // Check if fullCommand is "list"
+        else if (fullCommand.equals("list")) {
+            return new ListAllCommand();
```

Comment:
Should the else-if statements be on the same line as the closing parenthesis '}'?

Reply:
Indeed they should be! I have made the changes, thank you for pointing these out! :)